src/components/interaction-designer/block-editors/NameEditor.vue

```diff
@@ -63,16 +63,38 @@ export class NameEditor extends mixins(Lang) {
   }
 
   set blockName(value: IBlock['name']) {
-    this.block_setName({blockId: this.block.uuid, value})
+    this.block_setName({blockId: this.block.uuid, value, isUserDefinedName: true})
   }
 
   filterName(e: KeyboardEvent): void {
+    // Consider user input as name editing
+    this.editBlockName = true
+
     if (e.key.match(/\W+|Enter/g)) {
       e.preventDefault()
     }
   }
 
-  @flowVuexNamespace.Mutation block_setName!: ({blockId, value}: {blockId: IBlock['uuid'], value: IBlock['name']}) => void
+  handleCompleteEditing(): void {
+    this.editBlockName = false
+
+    if (this.blockName === '') {
+      // This will turn on automatic name generation from the label
+      this.block_resetName({blockId: this.block.uuid})
+    }
+  }
+
+  @flowVuexNamespace.Action block_setName!: ({
+    blockId,
+    value,
+    isUserDefinedName,
+  }: {
+    blockId: IBlock['uuid'],
+    value: IBlock['name'],
+    isUserDefinedName: IBlockUIMetadata['has_user_defined_name'],
```

Comment:
@object-required , can we get the type of `IBlockUIMetadata['has_user_defined_name']` if we didn't update flow-runner interface `IBlockUIMetadata` to have `has_user_defined_name` ?
Or did you add it but forgot to share the PR ?

Reply:
- [ ] update `IBlockUIMetadata` with the new flag (optional boolean)

Reply:
Added to https://github.com/FLOIP/flow-runner/pull/78